spring-cloud-stream/src/main/java/org/springframework/cloud/stream/config/ChannelBindingProperties.java

```diff
@@ -84,8 +113,120 @@ else if (binding instanceof Map) {
 		return channelName;
 	}
 
+	public int getPartitionCount(String channelName) {
+		Object binding = bindings.get(channelName);
+		// we may shortcut directly to the path
+		if (binding instanceof Map) {
+			try {
+				Map<?, ?> bindingProperties = (Map<?, ?>) binding;
+				Object bindingPath = bindingProperties.get(PARTITION_COUNT);
+				if (bindingPath != null) {
+					return Integer.parseInt(bindingPath.toString());
+				}
+			} catch (NumberFormatException e) {
+				// ignore and just return 1
+			}
+		}
+		return 1;
+	}
+
+	public boolean isPartitionedConsumer(String channelName) {
+		Object binding = bindings.get(channelName);
+		// if the setting is just a target shortcut
+		if (binding == null || binding instanceof String) {
+			return false;
+		}
+		else if (binding instanceof Map) {
+			Map<?, ?> bindingProperties = (Map<?, ?>) binding;
+			Object bindingPath = bindingProperties.get(PARTITIONED);
+			if (bindingPath != null) {
+				return Boolean.valueOf(bindingPath.toString());
+			}
+		}
+		// just return the channel name if not found
+		return false;
+	}
+
+	public boolean isPartitionedProducer(String channelName) {
+		Object binding = bindings.get(channelName);
+		// if the setting is just a target shortcut
+		if (binding == null || binding instanceof String) {
+			return false;
+		}
+		else if (binding instanceof Map) {
+			Map<?, ?> bindingProperties = (Map<?, ?>) binding;
+			return bindingProperties.get(BinderProperties.PARTITION_KEY_EXPRESSION) != null
+					|| bindingProperties.get(BinderProperties.PARTITION_KEY_EXTRACTOR_CLASS) != null;
+		}
+		return false;
+	}
+
+	/**
+	 * Merge general properties provided by 'spring.cloud.stream.consumerProperties.*' with individual binding
+	 * properties supplied via binders.
+	 *
+	 * @param inputChannelName the input channel name
+	 * @return merged consumer properties
+	 */
+	public Properties getConsumerProperties(String inputChannelName) {
+		if (isPartitionedConsumer(inputChannelName)) {
+			Properties channelConsumerProperties = new Properties();
+			if (consumerProperties != null) {
+				channelConsumerProperties.putAll(consumerProperties);
+			}
+			channelConsumerProperties.setProperty(BinderProperties.COUNT,
```

Comment:
I think we need to make the distinction between `minPartitionCount` and `partitionCount` at some point of time (especially for Kafka). Otherwise, the min partition logic may not be used at all. What do you think?